metapackages/auto-instrumentations-node/src/utils.ts

```diff
@@ -253,3 +262,56 @@ export function getResourceDetectorsFromEnv(): Array<Detector | DetectorSync> {
     return resourceDetector || [];
   });
 }
+
+type PropagatorFactoryFunction = () => TextMapPropagator;
+
+const propagatorMap = new Map<string, PropagatorFactoryFunction>([
+  ['tracecontext', () => new W3CTraceContextPropagator()],
+  ['baggage', () => new W3CTraceContextPropagator()],
+  [
+    'b3',
+    () => new B3Propagator({ injectEncoding: B3InjectEncoding.SINGLE_HEADER }),
+  ],
+  [
+    'b3multi',
+    () => new B3Propagator({ injectEncoding: B3InjectEncoding.MULTI_HEADER }),
+  ],
+  ['jaeger', () => new JaegerPropagator()],
+  ['xray', () => new AWSXRayPropagator()],
+  ['ottrace', () => new OTTracePropagator()],
+]);
+
+/**
+ * Get a propagator based on the OTEL_PROPAGATORS env var.
+ */
+export function getPropagator(): TextMapPropagator {
+  if (process.env.OTEL_PROPAGATORS == null) {
```

Comment:
Should empty string be treated as "use the default" as well? (That's what I infer from the language in this section: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md#enum-value)

Also the spec mentions handling "none" as a value to say "no propagators": https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md#general-sdk-configuration

Reply:
>  Should empty string be treated as "use the default" as well? (That's what I infer from the language in this section: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md#enum-value)

Yes that's correct, good catch :slightly_smiling_face: I added that to the default case. https://github.com/open-telemetry/opentelemetry-js-contrib/pull/2232/commits/cf3bb0d733fd56c2daa0ec183c62be28b8a34743

> Also the spec mentions handling "none" as a value to say "no propagators": https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md#general-sdk-configuration

Also good catch, it would've incorrectly logged that it's unkown before (the Propgator would still have been a no-op CompositePropagator with no sub-propgators, so I think that would've been spec compliant). I added an info log that lets the user know that `none`. Though I'm not sure what to do if there's `none` among other valid values so I've opted to still take the rest of the propagators in that case. My guess is that users would be least surpirsed if they set `tracecontext, none` and the `W3CTraceContextPropagator` would be the one that gets set. https://github.com/open-telemetry/opentelemetry-js-contrib/pull/2232/commits/cf3bb0d733fd56c2daa0ec183c62be28b8a34743

Reply:
Some comparison points on "none" handling:
- auto-instrumentations-node handling of `OTEL_NODE_RESOURCE_DETECTORS` handles `foo,none,bar` and "none" https://github.com/open-telemetry/opentelemetry-js-contrib/blob/caf7cb503760d17ac6455dcf712f677ef665f07a/metapackages/auto-instrumentations-node/src/utils.ts#L238-L244
- ditto for sdk-node's handling of that same envvar
- sdk-node's handling of `OTEL_TRACES_EXPORTER` is slightly weird: `OTEL_TRACES_EXPORTER=none,otlp` results in different behaviour to `OTEL_TRACES_EXPORTER=otlp,none` :) https://github.com/open-telemetry/opentelemetry-js/blob/main/experimental/packages/opentelemetry-sdk-node/src/TracerProviderWithEnvExporter.ts#L101-L126 

Anyway, this is something that could perhaps share some common processing at some point. Doesn't have to be this PR, and perhaps would wait for work related to supporting file-config.